ingestion/src/metadata/ingestion/source/objectstore/s3/metadata.py

```diff
@@ -0,0 +1,287 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""S3 object store extraction metadata"""
+import json
+import random
+import traceback
+from datetime import datetime, timedelta
+from enum import Enum
+from typing import Iterable, Optional, List
+
+from metadata.generated.schema.entity.data.table import Column
+from metadata.ingestion.source.database.datalake.models import DatalakeColumnWrapper
+from pandas import DataFrame
+
+from metadata.generated.schema.entity.data import container
+from pydantic import Extra, Field, ValidationError
+from pydantic.main import BaseModel
+
+from metadata.generated.schema.api.data.createContainer import CreateContainerRequest
+from metadata.generated.schema.entity.data.container import ContainerDataModel
+from metadata.generated.schema.entity.services.connections.metadata.openMetadataConnection import (
+    OpenMetadataConnection,
+)
+from metadata.generated.schema.entity.services.connections.objectstore.s3ObjectStoreConnection import (
+    S3StoreConnection,
+)
+from metadata.generated.schema.metadataIngestion.objectstore.containerMetadataConfig import ObjectStoreContainerConfig, \
+    MetadataEntry
+from metadata.generated.schema.metadataIngestion.workflow import (
+    Source as WorkflowSource,
+)
+from metadata.ingestion.api.source import InvalidSourceException
+from metadata.ingestion.source.database.datalake.metadata import DatalakeSource
+from metadata.ingestion.source.objectstore.objectstore_service import (
+    ObjectStoreServiceSource,
+)
+from metadata.utils.filters import filter_by_container
+from metadata.utils.logger import ingestion_logger
+from metadata.utils.s3_utils import prefix_exits
+
+logger = ingestion_logger()
+
+OPENMETADATA_TEMPLATE_FILE_NAME = "openmetadata.json"
+
+
+class S3Metric(Enum):
+    NUMBER_OF_OBJECTS = "NumberOfObjects"
+    BUCKET_SIZE_BYTES = "BucketSizeBytes"
+
+
+class S3BucketResponse(BaseModel):
+    """
+    Class modelling a response received from s3_client.list_buckets operation
+    """
+
+    class Config:
+        extra = Extra.forbid
+
+    Name: str = Field(..., description="Bucket name", title="Bucket Name")
+    CreationDate: datetime = Field(
```

Comment:
We could stick to Python naming conventions, such as `name` or `creation_date`

Reply:
Updated naming to snake case for all BaseModels